src/nvim/edit.c

```diff
@@ -3386,11 +3386,23 @@ static bool ins_bs(int c, int mode, int *inserted_space_p)
         // again when auto-formatting.
         if (has_format_option(FO_AUTO)
             && has_format_option(FO_WHITE_PAR)) {
-          char *ptr = ml_get_buf_mut(curbuf, curwin->w_cursor.lnum);
+          const char *ptr = ml_get_buf(curbuf, curwin->w_cursor.lnum);
           int len = get_cursor_line_len();
           if (len > 0 && ptr[len - 1] == ' ') {
-            ptr[len - 1] = NUL;
-            curbuf->b_ml.ml_line_len--;
+            char *newp = xmalloc((size_t)(curbuf->b_ml.ml_line_textlen - 1));
+
+            memmove(newp, ptr, (size_t)(len - 1));
+            newp[len - 1] = NUL;
+            if (curbuf->b_ml.ml_line_textlen > len + 1) {
```

Comment:
`curbuf->b_ml.ml_line_textlen` and `len + 1` are always the same value here in Nvim.